packages/smooth_app/lib/pages/user_preferences_page.dart

```diff
@@ -227,11 +218,389 @@ class UserPreferencesPage extends StatelessWidget {
 
   List<AttributeGroup> _reorderGroups(List<AttributeGroup> groups) {
     final List<AttributeGroup> result = <AttributeGroup>[];
-    for (final String id in _ORDERED_ATTRIBUTE_GROUP_IDS) {
+    for (final String id in UserPreferencesPage._ORDERED_ATTRIBUTE_GROUP_IDS) {
       result.addAll(groups.where((AttributeGroup g) => g.id == id));
     }
-    result.addAll(groups.where(
-        (AttributeGroup g) => !_ORDERED_ATTRIBUTE_GROUP_IDS.contains(g.id)));
+    result.addAll(groups.where((AttributeGroup g) =>
+        !UserPreferencesPage._ORDERED_ATTRIBUTE_GROUP_IDS.contains(g.id)));
     return result;
   }
+
+  Widget _getColorButton(
+    final ColorScheme colorScheme,
+    final String colorTag,
+    final ThemeProvider themeProvider,
+  ) =>
+      TextButton(
+        onPressed: () async => themeProvider.setColorTag(colorTag),
+        style: TextButton.styleFrom(
+          backgroundColor: SmoothTheme.getColor(
+            colorScheme,
+            SmoothTheme.MATERIAL_COLORS[colorTag]!,
+            ColorDestination.BUTTON_BACKGROUND,
+          ),
+        ),
+        child: Icon(
+          Icons.palette,
+          color: SmoothTheme.getColor(
+            colorScheme,
+            SmoothTheme.MATERIAL_COLORS[colorTag]!,
+            ColorDestination.BUTTON_FOREGROUND,
+          ),
+        ),
+      );
+
+  String _getPreferenceFlagKey(final int groupIndex) {
+    switch (groupIndex) {
+      case 0:
+        return 'profile';
+      case 1:
+        return 'food';
+      case 2:
+        return 'vrac';
```

Comment:
Had to use the translator to understand this, would be good to stick with english